islandora_transforms/slurp_all_MODS_to_solr.xslt

```diff
@@ -1,245 +1,174 @@
-        <xsl:with-param name="pid" select="$pid"/>
-        <xsl:with-param name="datastream" select="$datastream"/>
-      </xsl:call-template>
-    </xsl:variable>
-
-    <!-- Use attributes in field name. -->
-    <xsl:variable name="this_prefix">
-      <xsl:value-of select="$prefix"/>
-      <xsl:for-each select="@*">
-        <xsl:sort select="concat(local-name(), namespace-uri(self::node()))"/>
-        <xsl:value-of select="local-name()"/>
-        <xsl:text>_</xsl:text>
-        <xsl:value-of select="."/>
-        <xsl:text>_</xsl:text>
-      </xsl:for-each>
-    </xsl:variable>
-
-    <!-- Prevent multiple generating multiple instances of single-valued fields
-         by tracking things in a HashSet -->
-    <xsl:variable name="field_name" select="normalize-space(concat($this_prefix, local-name()))"/>
-    <!-- The method java.util.HashSet.add will return false when the value is
-         already in the set. -->
-    <xsl:if test="java:add($single_valued_hashset, $field_name)">
-      <xsl:if test="not(normalize-space($textValue)='')">
-        <field>
-          <xsl:attribute name="name">
-            <xsl:value-of select="concat($field_name, '_dt')"/>
-          </xsl:attribute>
-          <xsl:value-of select="$textValue"/>
-        </field>
-      </xsl:if>
-      <xsl:if test="not(normalize-space($rawTextValue)='')">
-        <field>
-          <xsl:attribute name="name">
-            <xsl:value-of select="concat($field_name, '_s')"/>
-          </xsl:attribute>
-          <xsl:value-of select="$rawTextValue"/>
-        </field>
-      </xsl:if>
-    </xsl:if>
-
-    <xsl:if test="not(normalize-space($textValue)='')">
-      <field>
-        <xsl:attribute name="name">
-          <xsl:value-of select="concat($prefix, local-name(), '_mdt')"/>
-        </xsl:attribute>
-        <xsl:value-of select="$textValue"/>
-      </field>
-    </xsl:if>
-    <xsl:if test="not(normalize-space($rawTextValue)='')">
-      <field>
-        <xsl:attribute name="name">
-          <xsl:value-of select="concat($prefix, local-name(), '_ms')"/>
-        </xsl:attribute>
-        <xsl:value-of select="$rawTextValue"/>
-      </field>
-    </xsl:if>
+  <!-- We pull out cla topics for the visualization index -->
+  <xsl:template match="mods:subject[@authority='cla_topic']" mode="CollectingLA">
+    <!-- cla_topic_mt field is created automatically from submitted _ms one -->
+    <field name="cla_topic_ms">
+      <xsl:value-of select="mods:topic"/>
+    </field>
+  </xsl:template>
+  
+  <xsl:template match="mods:originInfo[mods:dateCreated[@encoding='iso8601']]" mode="CollectingLA">
+    <xsl:variable name="dateStart"
+      select="java:edu.ucla.library.IsoToSolrDateConverter.getStartDateFromIsoDateString(normalize-space(mods:dateCreated[@encoding='iso8601']))" />
+    <xsl:variable name="dateEnd"
+      select="java:edu.ucla.library.IsoToSolrDateConverter.getEndDateFromIsoDateString(normalize-space(mods:dateCreated[@encoding='iso8601']))" />
```

Comment:
We're content to go with the DGI one now. Our doesn't do anything that yours doesn't and yours puts a placeholder in for unrecognized dates, which is a nice feature.